front/src/reducers/osrdconf2/common/index.ts

```diff
@@ -0,0 +1,294 @@
+      state.rollingStockComfort = action.payload;
+    },
+    updateSpeedLimitByTag(state: Draft<S>, action: PayloadAction<S['speedLimitByTag']>) {
+      state.speedLimitByTag = action.payload;
+    },
+    updateOrigin(state: Draft<S>, action: PayloadAction<S['origin']>) {
+      state.origin = action.payload;
+    },
+    updateInitialSpeed(state: Draft<S>, action: PayloadAction<S['initialSpeed']>) {
+      state.initialSpeed = action.payload;
+    },
+    updateDepartureTime(state: Draft<S>, action: PayloadAction<S['departureTime']>) {
+      state.departureTime = action.payload;
+    },
+    updateOriginTime(state: Draft<S>, action: PayloadAction<S['originTime']>) {
+      if (action.payload) {
+        const newOriginTimeSeconds = time2sec(action.payload);
+        const { originLinkedBounds, originTime, originUpperBoundTime } = state;
+        if (originLinkedBounds) {
+          const difference =
+            originTime && originUpperBoundTime
+              ? time2sec(originUpperBoundTime) - time2sec(originTime)
+              : ORIGIN_TIME_BOUND_DEFAULT_DIFFERENCE;
+          state.originUpperBoundTime = sec2time(newOriginTimeSeconds + difference);
+        }
+        if (
+          state.originUpperBoundTime &&
+          time2sec(action.payload) > time2sec(state.originUpperBoundTime)
+        ) {
+          state.originTime = state.originUpperBoundTime;
+        } else {
+          state.originTime = action.payload;
+        }
+      }
+    },
+    updateOriginUpperBoundTime(state: Draft<S>, action: PayloadAction<S['originUpperBoundTime']>) {
+      if (action.payload) {
+        const newOriginUpperBoundTimeSeconds = time2sec(action.payload);
+        if (state.originLinkedBounds) {
+          const difference =
+            state.originTime && state.originUpperBoundTime
+              ? time2sec(state.originUpperBoundTime) - time2sec(state.originTime)
+              : ORIGIN_TIME_BOUND_DEFAULT_DIFFERENCE;
+          state.originTime = sec2time(newOriginUpperBoundTimeSeconds - difference);
+        }
+        if (state.originTime && time2sec(action.payload) < time2sec(state.originTime)) {
+          state.originUpperBoundTime = state.originTime;
+        } else {
+          state.originUpperBoundTime = action.payload;
+        }
+      }
+    },
+    toggleOriginLinkedBounds(state: Draft<S>) {
+      state.originLinkedBounds = !state.originLinkedBounds;
+    },
+    updateOriginDate(state: Draft<S>, action: PayloadAction<S['originDate']>) {
+      state.originDate = action.payload;
+    },
+    updateOriginUpperBoundDate(state: Draft<S>, action: PayloadAction<S['originUpperBoundDate']>) {
+      state.originUpperBoundDate = action.payload;
+    },
+    replaceVias(state: Draft<S>, action: PayloadAction<S['vias']>) {
+      state.vias = action.payload;
+    },
+    // TODO rename updateVias -> addVias
+    addVias(state: Draft<S>, action: PayloadAction<PointOnMap>) {
```

Comment:
Hum if we rename it here, then we also need to change the osrdconf reducer, otherwise this will break when we will switch to this store slice.
So 2 possibilities here : 
- rename also in osrdconf now, rename all the usages (only 1-2 I think) and remove the comment line 212
- rename back to updateVias in this slice, and leave the comment so we'll do the rename later

Personally, I think we can do the rename later, but the choice is yours. Also, as you can add only 1 via at the time, I think addVia without S would be a better name.

```suggestion
    addVias(state: Draft<S>, action: PayloadAction<PointOnMap>) {
```

Reply:
For now, I keep updateVias, I will rename it when I will switch to store slice.